### PR TITLE
Data proxy cleanup

### PR DESCRIPTION
**Pull request by Thomasvdam**
## Motivation

Small cleanup removing some nullable proto annotations as they didn't seem to add any value but made the code a little more difficult to deal with.

As discussed we prefer not to interact with the stores directly when it's not from the immediate Keeper context. This means most get/set/has operations need their own wrapper so the boundaries that expose the Keeper don't need to have knowledge of the actual store implementation.

## Explanation of Changes

I didn't find a way to cleanly prevent the boundaries (abci.go, msg_server.go, grpc_query.go) from accessing the stores directly, but at least most operations now have their own API and the boundary doesn't need to know anything about how the store works internally.

## Testing

Verify tests still pass

## Related PRs and Issues

Should be the last part of #316

### CODE REVIEW

File: x/data-proxy/keeper/keeper.go [hunk 1]
```diff
@@ -49,72 +48,87 @@ func (k Keeper) GetAuthority() string {
...
+
+	itr, err := k.feeUpdateQueue.Iterate(ctx, rng)
+	if err != nil {
+		return nil, err
+	}
+
+	keys, err := itr.Keys()
+	if err != nil {
+		return nil, err
+	}
```
Comment (gluax) [inline x/data-proxy/keeper/keeper.go:119]: Can `Go` steal `rust`'s `?` operator already 😢

**Review by gluax — APPROVED** [review]

File: x/data-proxy/keeper/params.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sedaprotocol/seda-chain/x/data-proxy/types"
+)
+
+func (k Keeper) StoreParams(ctx sdk.Context, params types.Params) error {
```
Comment (hacheigriega) [inline x/data-proxy/keeper/params.go:9]: `SetParams` just to follow SDK convention?

File: x/data-proxy/keeper/keeper.go [hunk 3]
```diff
@@ -49,72 +48,87 @@ func (k Keeper) GetAuthority() string {
...
-func (k Keeper) GetDataProxyConfig(ctx context.Context, pubKey string) (result types.ProxyConfig, err error) {
-	pubKeyBytes, err := hex.DecodeString(pubKey)
-	if err != nil {
-		return types.ProxyConfig{}, err
-	}
+func (k Keeper) HasDataProxy(ctx sdk.Context, pubKey []byte) (bool, error) {
+	return k.dataProxyConfigs.Has(ctx, pubKey)
+}
+
+func (k Keeper) StoreDataProxyConfig(ctx context.Context, pubKey []byte, proxyConfig types.ProxyConfig) error {
```
Comment (hacheigriega) [inline x/data-proxy/keeper/keeper.go:55]: `SetDataProxyConfig`

File: x/data-proxy/keeper/keeper.go [hunk 4]
```diff
@@ -49,72 +48,87 @@ func (k Keeper) GetAuthority() string {
...
-
-	keys, err := itr.Keys()
-	if err != nil {
-		return nil, err
-	}
-
-	for _, k := range keys {
-		pubkeys = append(pubkeys, k.K2())
-	}
+func (k Keeper) StoreFeeUpdate(ctx sdk.Context, height int64, pubKey []byte) error {
```
Comment (hacheigriega) [inline x/data-proxy/keeper/keeper.go:68]: `SetFeeUpdate` just to follow SDK convention

**Review by hacheigriega — APPROVED** [review]

**Pull request merged**